activesupport/lib/active_support/cache/file_store.rb

```diff
@@ -74,7 +74,7 @@ def delete_matched(matcher, options = nil)
       private
         def read_entry(key, **options)
           if File.exist?(key)
-            entry = File.open(key) { |f| Marshal.load(f) }
+            entry = File.open(key) { |f| @coder.load(f.read) }
```

Comment:
Should we go through `deserialize_entry` here instead of using `@coder` directly?

```suggestion
            entry = File.open(key) { |f| deserialize_entry(f.read) }
```